libevm/VM.cpp

```diff
@@ -1096,7 +1492,9 @@ void VM::interpretCases()
 #if EVM_HACK_DUP_64
 			*(uint64_t*)m_SPP = *(uint64_t*)(m_SP + n);
 #else
-			m_SPP[0] = m_SP[n];
+			// the stack slot being copied into may no longer hold a u256
+			// so we construct a new one in the memory, rather than assign
+			new(m_SPP) u256(m_SP[n]);
```

Comment:
It's not the source that is the problem - it's a u256.  It's the destination that's the problem - it might be a vector.  The assignment operator would treat the destination as a u256, which might blow up.  Placement new treats it as raw memory.  (Then we are just down to the problem that the destination's destructor is not called.  I think we will need XPOP and XPUSH to fix that.)